core/src/main/scala/doric/types/SparkType.scala

```diff
@@ -79,6 +79,8 @@ object SparkType {
     override def dataType: DataType = dt
   }
 
+  implicit val fromNull: SparkType[Null] = SparkType[Null](NullType)
```

Comment:
Add castings to Null type, in my oppinon all can be safe castings, it will always work that transformation and allows to a better sintax in some cases.